### PR TITLE
In-Memory Caching In C#

### LINKED ISSUE #5744: In-Memory Caching In C#
### Topic Suggestion
In-Memory Caching In C#

### Writing sample(s): 
https://www.section.io/engineering-education/java-timer-tasks/
https://www.section.io/engineering-education/creating-a-blazor-server-side-application-using-aspdotnet-core-to-perform-crud-operations/

#### Proposed title of article
In-Memory Caching In C#

#### Proposed article introduction
During app development, application performance is important. The fact that your application works well during app development doesn't mean it will work well when thousands of people start using it. One main area to look at is the data access. Toking your data can be very expansive especially when your data might not change very often. This is where data caching comes in. This data can be cached in a number of ways that we do not indicate going back to the data base. 

In this article, I shall be looking at a caching method in .NET called in-memory caching in C#. I will create an application without caching, observe its performance and then introduce caching and observe how the performance is improved.

#### Key takeaways 
- Understand the concept of in-memory caching
- Understand how to implement in-memory caching using C#
- Understand the concept of .net in our app development

#### Article quality
In this article, I will be creating an app, observe its performance without caching, implement caching in our app and observe the performance.

#### References
N/A

### PR DESCRIPTION
**Pull request by Joesongoma**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [X] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [X] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [X] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #5744

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by Joesongoma** [comment]
@Dawe-7
If you are available, could you please give my article a review. Thanks.

**Comment by Kayere** [comment]
Plagiarism check done ✅

File: content/articles/in-memory-caching-in-c#/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+images:
+
+  - url: /engineering-education/in-memory-caching-in-c#/hero.jpg
+    alt: In-Memory Caching In C# Hero Image
+---
+Caching is used to store data from slow memory to a faster memory to enhance easy access of data in a website. During app development, application performance is important. 
+<!--more-->
+The fact that your application works well during app development does not mean it will work well when thousands of people start using it. One main area to look at is data access. 
+
+Talking your database can be very expansive especially when your data might not change very often. This is where data caching comes in. This data can be cached in several ways such that we do not need to go back to the database to access it.
```
Comment (Dawe-7) [inline content/articles/in-memory-caching-in-c#/index.md:21]: This first sentence here does not make sense, kindly change it or rephrase it.

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
Nice article @Joesongoma. As i was doing a final grammar and tech review before marking the article as `ready for final review` i noticed a minor grammatical error, kindly make the change and your article will be good to go.

**Comment by Joesongoma** [comment]
@Dawe-7, Requested changes made. Thanks

**Review by Dawe-7 — APPROVED** [review]
Nice article @Joesongoma , kindly go through the changes made throughout the review process so as to improve on the flow and format of your future articles. Thank you.

**Comment by Kayere** [comment]
Good article @Joesongoma - be sure to cross-check the image links before submitting your article for review. Otherwise, great job on the article.

cc @Dawe-7

**Pull request merged**